controlplane/kubeadm/api/v1alpha3/condition_consts.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+// Conditions and condition Reasons for the KubeadmControlPlane object
+
+const (
+	// MachinesReady reports an aggregate of current status of the machines controlled by the KubeadmControlPlane.
+	MachinesReadyCondition clusterv1.ConditionType = "MachinesReady"
+)
+
+const (
+	// CertificatesGeneratedCondition documents that cluster certificates were generated as part of the
+	// processing of a a KubeadmControlPlane object.
+	// IMPORTANT: This condition won't be re-created after clusterctl move.
+	CertificatesGeneratedCondition clusterv1.ConditionType = "CertificatesGenerated"
+)
+
+const (
+	// InitializedCondition documents that the control plane was initialized and the first control plane instance
+	// has completed the kubeadm init operation; as a consequence, there is an API server instance ready for
+	// processing requests.
+	InitializedCondition clusterv1.ConditionType = "Initialized"
```

Comment:
@vincepri I have added this condition as agreed, but I think that what we really need is intead is control plane ready (name of the condition TBD)

Reply:
We probably need both? Ready and Initialized

Reply:
I think it depends on what we want things to mean semantically.

Initialized has value today in that we are triggering on it as part of the contract for backwards compatibility reasons. 

From an external point of view, I think users really only care about Availability and if it is quiescent (how the CAEP defines Ready).

I think `Initialized` could be valuable in exposing progression towards quiescence.

Reply:
I'm -1 for both.
IMO the users are interested in knowing when the API server is available e.g. to apply addons, but I don't think there is value in reporting when the kubeadm config is initalized

Reply:
Today, the Cluster object has both `ControlPlaneInitialized` and `ControlPlaneReady` without a clear definition of what those mean, let's take this opportunity to better define these based on what a user needs to see, and our controllers.

Initialized has been used often in controllers to understand if we can try to connect to the control plane after the first api-server machine is up. I do agree that maybe we're looking to express this a little better, for example:

- `Available`: The control plane is available to receive requests.
- `Ready`: The control plane is ready, every node in the control plane is also ready (needs CNI).

Initialized can be replaced by available, and we can gradually migrate controllers in the future to rely on that condition instead.

Reply:
Ok for Available